src/core/jpx.js

```diff
@@ -119,13 +124,14 @@ class JpxImage {
     try {
       const size = bytes.length;
       ptr = module._malloc(size);
-      module.HEAPU8.set(bytes, ptr);
+      module.writeArrayToMemory(bytes, ptr);
       const ret = module._jp2_decode(
         ptr,
         size,
         numComponents > 0 ? numComponents : 0,
         !!isIndexedColormap,
-        !!smaskInData
+        !!smaskInData,
+        Math.max(0, Math.floor(reducePower ?? 0))
```

Comment:
Can this be simplified, since the default value is zero (above) and it seems (as far as I can tell) that `getReducePowerForJPX` will always return a positive integer?